.github/workflows/release.yml

```diff
@@ -16,7 +16,8 @@ permissions:
   contents: read
 jobs:
   release:
-    runs-on: macos-12
+    # macos-14 and later does not seem to pass the reproducibility check?
+    runs-on: macos-13-large
```

Comment:
Still fails 😞 
https://github.com/lima-vm/socket_vmnet/actions/runs/11943937070/job/33293946939

```
diffoscope _artifacts.0/ _artifacts.1/
--- _artifacts.0/
+++ _artifacts.1/
│   --- _artifacts.0/SHA256SUMS
├── +++ _artifacts.1/SHA256SUMS
│ @@ -1,3 +1,3 @@
│  17933034e4fd9b855ffa92d29181695fe57ff853480406532ca220e3d24797a1  build-env.txt
│ -8306c566e23d2587b509d8804854cd855871b374146ea55c1a3d19b024008732  socket_vmnet-1.1.7-32-g64b35f0-arm64.tar.gz
│ -6889e78fe91c104fb38c3050c0272598a967c8ef4f91953860bfaad55b4313b4  socket_vmnet-1.1.7-32-g64b35f0-x86_64.tar.gz
│ +2ecd76cba9d0c3b75fa0be3a4bae1331e82ad0e51848eb25a4f3047beb39daf7  socket_vmnet-1.1.7-32-g64b35f0-arm64.tar.gz
│ +eff39669f67d1af08de9c14227af6c12cbca0465b092d8da7ec8d64b63172662  socket_vmnet-1.1.7-32-g64b35f0-x86_64.tar.gz
│   --- _artifacts.0/socket_vmnet-1.1.7-32-g64b35f0-arm64.tar.gz
├── +++ _artifacts.1/socket_vmnet-1.1.7-32-g64b35f0-arm64.tar.gz
│ ├── socket_vmnet-1.1.7-32-g64b35f0-arm64.tar
│ │ ├── ./opt/socket_vmnet/bin/socket_vmnet
│ │ │ ├── arm64
│ │ │ │┄ Format-specific differences are supported for this file format but no file-specific differences were detected; falling back to a binary diff. file(1) reports: Mach-O 64-bit arm64 executable, flags:<NOUNDEFS|DYLDLINK|TWOLEVEL|PIE>
│ │ │ │ @@ -74,16 +74,16 @@
│ │ │ │  00000490: 0b00 0000 5000 0000 0000 0000 0100 0000  ....P...........
│ │ │ │  000004a0: 0100 0000 0100 0000 0200 0000 5d00 0000  ............]...
│ │ │ │  000004b0: 0000 0000 0000 0000 0000 0000 0000 0000  ................
│ │ │ │  000004c0: 0000 0000 0000 0000 f88d 0000 a500 0000  ................
│ │ │ │  000004d0: 0000 0000 0000 0000 0000 0000 0000 0000  ................
│ │ │ │  000004e0: 0e00 0000 2000 0000 0c00 0000 2f75 7372  .... ......./usr
│ │ │ │  000004f0: 2f6c 6962 2f64 796c 6400 0000 0000 0000  /lib/dyld.......
│ │ │ │ -00000500: 1b00 0000 1800 0000 6080 fdd9 78c3 37dc  ........`...x.7.
│ │ │ │ -00000510: 8ed9 6982 a9c1 f826 3200 0000 2000 0000  ..i....&2... ...
│ │ │ │ +00000500: 1b00 0000 1800 0000 f9a5 df07 d58f 354b  ..............5K
│ │ │ │ +00000510: ab55 c674 4280 7ed1 3200 0000 2000 0000  .U.tB.~.2... ...
│ │ │ │  00000520: 0100 0000 0000 0d00 0002 0e00 0100 0000  ................
│ │ │ │  00000530: 0300 0000 0001 fe03 2a00 0000 1000 0000  ........*.......
│ │ │ │  00000540: 0000 0000 0000 0000 2800 0080 1800 0000  ........(.......
│ │ │ │  00000550: 6c13 0000 0000 0000 0000 0000 0000 0000  l...............
│ │ │ │  00000560: 0c00 0000 5800 0000 1800 0000 0200 0000  ....X...........
│ │ │ │  00000570: 0000 0100 0000 0100 2f53 7973 7465 6d2f  ......../System/
│ │ │ │  00000580: 4c69 6272 6172 792f 4672 616d 6577 6f72  Library/Framewor
│ │ │ │ @@ -2407,17 +2407,17 @@
│ │ │ │  00009660: fade 0cc0 0000 01b9 0000 0001 0000 0000  ................
│ │ │ │  00009670: 0000 0014 fade 0c02 0000 01a5 0002 0400  ................
│ │ │ │  00009680: 0002 0002 0000 0065 0000 0058 0000 0000  .......e...X....
│ │ │ │  00009690: 0000 000a 0000 9660 2002 000c 0000 0000  .......` .......
│ │ │ │  000096a0: 0000 0000 0000 0000 0000 0000 0000 0000  ................
│ │ │ │  000096b0: 0000 0000 0000 0000 0000 0000 0000 0000  ................
│ │ │ │  000096c0: 0000 4000 0000 0000 0000 0001 736f 636b  ..@.........sock
│ │ │ │ -000096d0: 6574 5f76 6d6e 6574 00d6 87c0 818d f411  et_vmnet........
│ │ │ │ -000096e0: 9acd f456 ad8a 1bff f09b 16f8 a164 50a8  ...V.........dP.
│ │ │ │ -000096f0: abb3 7332 213a 4922 0e25 6a6b 7913 9767  ..s2!:I".%jky..g
│ │ │ │ +000096d0: 6574 5f76 6d6e 6574 0009 2b2a 5dfd 3f4c  et_vmnet..+*].?L
│ │ │ │ +000096e0: fd5d c210 6909 9a56 2c5c ac3e 4240 adef  .]..i..V,\.>B@..
│ │ │ │ +000096f0: c2e1 5e7c 4f7b 86ae 2625 6a6b 7913 9767  ..^|O{..&%jky..g
│ │ │ │  00009700: 16d7 a55e eaa5 885f cf14 322e e27f ce76  ...^..._..2....v
│ │ │ │  00009710: ecc2 4284 59d2 3f3e fea8 01a9 dc02 fd17  ..B.Y.?>........
│ │ │ │  00009720: 0f35 1404 72cd 91d3 1dda d99b 638b e478  .5..r.......c..x
│ │ │ │  00009730: aa88 91a4 2023 8b16 e7ce 6fa2 832d 7cff  .... #....o..-|.
│ │ │ │  00009740: 7edc e525 bcd6 287b 5e94 2d37 b350 0907  ~..%..({^.-7.P..
│ │ │ │  00009750: 3fad deab 2cc7 52da 5b9a dc53 4a27 1e6a  ?...,.R.[..SJ'.j
│ │ │ │  00009760: b062 6705 0d0d cf90 31bd 7864 3c18 77bd  .bg.....1.xd<.w.
│ │ ├── ./opt/socket_vmnet/bin/socket_vmnet_client
│ │ │ ├── arm64
│ │ │ │┄ Format-specific differences are supported for this file format but no file-specific differences were detected; falling back to a binary diff. file(1) reports: Mach-O 64-bit arm64 executable, flags:<NOUNDEFS|DYLDLINK|TWOLEVEL|PIE>
│ │ │ │ @@ -50,15 +50,15 @@
│ │ │ │  00000310: 0000 0000 0100 0000 0100 0000 0100 0000  ................
│ │ │ │  00000320: 0200 0000 0f00 0000 0000 0000 0000 0000  ................
│ │ │ │  00000330: 0000 0000 0000 0000 0000 0000 0000 0000  ................
│ │ │ │  00000340: 7082 0000 1c00 0000 0000 0000 0000 0000  p...............
│ │ │ │  00000350: 0000 0000 0000 0000 0e00 0000 2000 0000  ............ ...
│ │ │ │  00000360: 0c00 0000 2f75 7372 2f6c 6962 2f64 796c  ..../usr/lib/dyl
│ │ │ │  00000370: 6400 0000 0000 0000 1b00 0000 1800 0000  d...............
│ │ │ │ -00000380: a590 2563 a30d 31d6 bca0 16f7 8e9e f21d  ..%c..1.........
│ │ │ │ +00000380: 4e5e 09fb 97a0 367e 9ec6 2017 ef04 318c  N^....6~.. ...1.
│ │ │ │  00000390: 3200 0000 2000 0000 0100 0000 0000 0d00  2... ...........
│ │ │ │  000003a0: 0002 0e00 0100 0000 0300 0000 0001 fe03  ................
│ │ │ │  000003b0: 2a00 0000 1000 0000 0000 0000 0000 0000  *...............
│ │ │ │  000003c0: 2800 0080 1800 0000 403c 0000 0000 0000  (.......@<......
│ │ │ │  000003d0: 0000 0000 0000 0000 0c00 0000 3800 0000  ............8...
│ │ │ │  000003e0: 1800 0000 0200 0000 013d 3805 0000 0100  .........=8.....
│ │ │ │  000003f0: 2f75 7372 2f6c 6962 2f6c 6962 5379 7374  /usr/lib/libSyst
│ │ │ │ @@ -2108,16 +2108,16 @@
│ │ │ │  000083b0: 0000 0014 fade 0c02 0000 018c 0002 0400  ................
│ │ │ │  000083c0: 0002 0002 0000 006c 0000 0058 0000 0000  .......l...X....
│ │ │ │  000083d0: 0000 0009 0000 83a0 2002 000c 0000 0000  ........ .......
│ │ │ │  000083e0: 0000 0000 0000 0000 0000 0000 0000 0000  ................
│ │ │ │  000083f0: 0000 0000 0000 0000 0000 0000 0000 0000  ................
│ │ │ │  00008400: 0000 4000 0000 0000 0000 0001 736f 636b  ..@.........sock
│ │ │ │  00008410: 6574 5f76 6d6e 6574 5f63 6c69 656e 7400  et_vmnet_client.
│ │ │ │ -00008420: 6f60 8241 9869 b458 e9ed abf7 9097 892b  o`.A.i.X.......+
│ │ │ │ -00008430: 2b0f 477b 7c92 fc8b c3f9 c2cc 9445 d088  +.G{|........E..
│ │ │ │ +00008420: bdce e8b8 2b65 fad0 cda8 877b ef4e 4d5d  ....+e.....{.NM]
│ │ │ │ +00008430: c63e 9360 0626 9aa7 3296 2935 201e 4ed6  .>.`.&..2.)5 .N.
│ │ │ │  00008440: ad7f acb2 586f c6e9 66c0 04d7 d1d1 6b02  ....Xo..f.....k.
│ │ │ │  00008450: 4f58 05ff 7cb4 7c7a 85da bd8b 4889 2ca7  OX..|.|z....H.,.
│ │ │ │  00008460: ad7f acb2 586f c6e9 66c0 04d7 d1d1 6b02  ....Xo..f.....k.
│ │ │ │  00008470: 4f58 05ff 7cb4 7c7a 85da bd8b 4889 2ca7  OX..|.|z....H.,.
│ │ │ │  00008480: 7126 7ae4 06c7 2ac4 5eb6 b6aa 402a 44ee  q&z...*.^...@*D.
│ │ │ │  00008490: 7e81 2c61 ba58 15ff 27d6 7598 16e9 2447  ~.,a.X..'.u...$G
│ │ │ │  000084a0: 31e1 e56b ea37 0bd0 a252 b57d a39b 6f7b  1..k.7...R.}..o{
│ ├── stat {}
│ │ @@ -1 +1 @@
│ │ -16777220 12888368315 -rw-r--r-- 1 runner staff 0 19982 "Nov 21 00:06:15 2024" "Nov 21 00:06:15 2024" "Nov 21 00:06:54 2024" "Nov 21 00:06:15 2024" 4096 40 0 _artifacts.0/socket_vmnet-1.1.7-32-g64b35f0-arm64.tar.gz
│ │ +16777220 12888368392 -rw-r--r-- 1 runner staff 0 19981 "Nov 21 00:06:15 2024" "Nov 21 00:06:15 2024" "Nov 21 00:06:55 2024" "Nov 21 00:06:15 2024" 4096 40 0 _artifacts.1/socket_vmnet-1.1.7-32-g64b35f0-arm64.tar.gz
│   --- _artifacts.0/socket_vmnet-1.1.7-32-g64b35f0-x86_64.tar.gz
├── +++ _artifacts.1/socket_vmnet-1.1.7-32-g64b35f0-x86_64.tar.gz
│ ├── socket_vmnet-1.1.7-32-g64b35f0-x86_64.tar
│ │ ├── ./opt/socket_vmnet/bin/socket_vmnet
│ │ │ ├── x86_64
│ │ │ │┄ Format-specific differences are supported for this file format but no file-specific differences were detected; falling back to a binary diff. file(1) reports: Mach-O 64-bit x86_64 executable, flags:<NOUNDEFS|DYLDLINK|TWOLEVEL|PIE>
│ │ │ │ @@ -69,16 +69,16 @@
│ │ │ │  00000440: 0b00 0000 5000 0000 0000 0000 0100 0000  ....P...........
│ │ │ │  00000450: 0100 0000 0100 0000 0200 0000 5d00 0000  ............]...
│ │ │ │  00000460: 0000 0000 0000 0000 0000 0000 0000 0000  ................
│ │ │ │  00000470: 0000 0000 0000 0000 f88d 0000 a500 0000  ................
│ │ │ │  00000480: 0000 0000 0000 0000 0000 0000 0000 0000  ................
│ │ │ │  00000490: 0e00 0000 2000 0000 0c00 0000 2f75 7372  .... ......./usr
│ │ │ │  000004a0: 2f6c 6962 2f64 796c 6400 0000 0000 0000  /lib/dyld.......
│ │ │ │ -000004b0: 1b00 0000 1800 0000 6d4d 69a4 0323 304a  ........mMi..#0J
│ │ │ │ -000004c0: 90ee 23c4 017f 9b58 3200 0000 2000 0000  ..#....X2... ...
│ │ │ │ +000004b0: 1b00 0000 1800 0000 b126 ff1a fc6e 386b  .........&...n8k
│ │ │ │ +000004c0: 9dcc a889 87b7 1a03 3200 0000 2000 0000  ........2... ...
│ │ │ │  000004d0: 0100 0000 0000 0d00 0002 0e00 0100 0000  ................
│ │ │ │  000004e0: 0300 0000 0001 fe03 2a00 0000 1000 0000  ........*.......
│ │ │ │  000004f0: 0000 0000 0000 0000 2800 0080 1800 0000  ........(.......
│ │ │ │  00000500: d012 0000 0000 0000 0000 0000 0000 0000  ................
│ │ │ │  00000510: 0c00 0000 5800 0000 1800 0000 0200 0000  ....X...........
│ │ │ │  00000520: 0000 0100 0000 0100 2f53 7973 7465 6d2f  ......../System/
│ │ │ │  00000530: 4c69 6272 6172 792f 4672 616d 6577 6f72  Library/Framewor
│ │ ├── ./opt/socket_vmnet/bin/socket_vmnet_client
│ │ │ ├── x86_64
│ │ │ │┄ Format-specific differences are supported for this file format but no file-specific differences were detected; falling back to a binary diff. file(1) reports: Mach-O 64-bit x86_64 executable, flags:<NOUNDEFS|DYLDLINK|TWOLEVEL|PIE>
│ │ │ │ @@ -50,15 +50,15 @@
│ │ │ │  00000310: 0000 0000 0100 0000 0100 0000 0100 0000  ................
│ │ │ │  00000320: 0200 0000 0f00 0000 0000 0000 0000 0000  ................
│ │ │ │  00000330: 0000 0000 0000 0000 0000 0000 0000 0000  ................
│ │ │ │  00000340: 7082 0000 1c00 0000 0000 0000 0000 0000  p...............
│ │ │ │  00000350: 0000 0000 0000 0000 0e00 0000 2000 0000  ............ ...
│ │ │ │  00000360: 0c00 0000 2f75 7372 2f6c 6962 2f64 796c  ..../usr/lib/dyl
│ │ │ │  00000370: 6400 0000 0000 0000 1b00 0000 1800 0000  d...............
│ │ │ │ -00000380: ff35 e2d1 5e9e 3a7b ab72 22da 2f3a 80bc  .5..^.:{.r"./:..
│ │ │ │ +00000380: b8a2 dc7c 1b54 3909 bbbd 19fa 3568 c3bc  ...|.T9.....5h..
│ │ │ │  00000390: 3200 0000 2000 0000 0100 0000 0000 0d00  2... ...........
│ │ │ │  000003a0: 0002 0e00 0100 0000 0300 0000 0001 fe03  ................
│ │ │ │  000003b0: 2a00 0000 1000 0000 0000 0000 0000 0000  *...............
│ │ │ │  000003c0: 2800 0080 1800 0000 703c 0000 0000 0000  (.......p<......
│ │ │ │  000003d0: 0000 0000 0000 0000 0c00 0000 3800 0000  ............8...
│ │ │ │  000003e0: 1800 0000 0200 0000 013d 3805 0000 0100  .........=8.....
│ │ │ │  000003f0: 2f75 7372 2f6c 6962 2f6c 6962 5379 7374  /usr/lib/libSyst
│ ├── stat {}
│ │ @@ -1 +1 @@
│ │ -16777220 12888368279 -rw-r--r-- 1 runner staff 0 19573 "Nov 21 00:06:15 2024" "Nov 21 00:06:15 2024" "Nov 21 00:06:54 2024" "Nov 21 00:06:15 2024" 4096 40 0 _artifacts.0/socket_vmnet-1.1.7-32-g64b35f0-x86_64.tar.gz
│ │ +16777220 12888368356 -rw-r--r-- 1 runner staff 0 19577 "Nov 21 00:06:15 2024" "Nov 21 00:06:15 2024" "Nov 21 00:06:55 2024" "Nov 21 00:06:15 2024" 4096 40 0 _artifacts.1/socket_vmnet-1.1.7-32-g64b35f0-x86_64.tar.gz
make: *** [test.repro] Error 1
```

Reply:
When did it start to fail? Can this be related to #65?

Reply:
Yes, let me revert #65